onnxruntime/core/session/inference_session.cc

```diff
@@ -178,6 +179,10 @@ static Status FinalizeSessionOptions(const SessionOptions& user_provided_session
 #if !defined(ORT_MINIMAL_BUILD)
   const logging::Logger& default_logger = logging::LoggingManager::DefaultLogger();
 
+  // Necessary otherwise INFO logs below won't actually be logged
+  auto defaultLoggerOrigSeverity = default_logger.GetSeverity();
+  logging::LoggingManager::SetDefaultLoggerSeverity(logging::Severity::kINFO);
```

Comment:
I believe not thread-safe in the sense that multiple threads could theoretically be setting severity and this would miss those changes. I don't know practically how much of an issue this is because at this point during session creation I am assuming there is only 1 thread involved. Note that the existing code has this issue where the INFO event is not actually being logged.
Overall not seeing the argument why this needs to be thread-safe, so if you can provide that reasoning that would be gr8 @snnn!